src/test/kotlin/test/ktortemplate/core/TestUtils.kt

```diff
@@ -1,96 +1,28 @@
-import org.koin.dsl.module
-import test.ktortemplate.conf.database.DatabaseConnection
-import test.ktortemplate.core.httphandler.defaultRoutes
-import test.ktortemplate.core.persistance.CarRepository
-import test.ktortemplate.core.persistance.PartRepository
-import test.ktortemplate.core.persistance.sql.CarMappingsTable
-import test.ktortemplate.core.persistance.sql.CarRepositoryImpl
-import test.ktortemplate.core.persistance.sql.PartMappingsTable
-import test.ktortemplate.core.persistance.sql.PartRepositoryImpl
-import test.ktortemplate.core.service.CarService
-import test.ktortemplate.core.service.CarServiceImpl
-import test.ktortemplate.core.utils.JsonSettings
-import javax.sql.DataSource
-
-private fun bootstrapDatabase(dbc: DatabaseConnection) {
-    dbc.query {
-        SchemaUtils.create(CarMappingsTable)
-        SchemaUtils.create(PartMappingsTable)
-    }
-}
-
-fun initServicesAndRepos() = module {
-    single<CarRepository> { CarRepositoryImpl() }
-    single<PartRepository> { PartRepositoryImpl() }
-    single<CarService> { CarServiceImpl() }
-}
-
-fun initDbCore() = module {
-    val dataSource: DataSource = HikariDataSource(
-        HikariConfig().apply {
-            driverClassName = "org.h2.Driver"
-            jdbcUrl = "jdbc:h2:mem:test"
-            maximumPoolSize = 5
-            isAutoCommit = true
-            transactionIsolation = "TRANSACTION_REPEATABLE_READ"
-            leakDetectionThreshold = 10000
-            poolName = "sat"
-            validate()
-        }
-    )
-
-    val dbc = DatabaseConnection(dataSource)
-    bootstrapDatabase(dbc)
-    single { dbc }
-}
-
-fun Application.testModule() {
-
-    install(DefaultHeaders)
-    install(Compression) {
-        gzip {
-            priority = 100.0
-        }
-        identity {
-            priority = 10.0
-        }
-        deflate {
-            priority = 1.0
-        }
-    }
-
-    install(CallLogging)
-    install(ContentNegotiation) {
-        register(ContentType.Application.Json, JacksonConverter(JsonSettings.mapper))
-    }
-    install(Routing) {
-        defaultRoutes()
-    }
-}
-
+import io.ktor.util.KtorExperimentalAPI
+import test.ktortemplate.containers.PgSQLContainerFactory
+import test.ktortemplate.module
+
+@KtorExperimentalAPI
+val testDatabaseConfigs: ApplicationConfig = MapApplicationConfig(
+    Pair("datasource.driver", PgSQLContainerFactory.instance.driverClassName),
+    Pair("datasource.jdbcUrl", PgSQLContainerFactory.instance.jdbcUrl),
+    Pair("datasource.username", PgSQLContainerFactory.instance.username),
+    Pair("datasource.password", PgSQLContainerFactory.instance.password),
+    Pair("datasource.pool.defaultIsolation", "TRANSACTION_REPEATABLE_READ"),
+    Pair("datasource.pool.maxPoolSize", "5")
+)
+
+@KtorExperimentalAPI
 fun <R> testApp(test: TestApplicationEngine.() -> R): R {
     return withTestApplication(
         {
-            testModule()
+            module(testDatabaseConfigs)
```

Comment:
And here is one of the big wins: we use the same application as in production. No risk of declaring a module here for the sake of making the tests pass and then forgetting about it in production. This has already happened. in the past.